### ISSUE TITLE
proj4 unable to install and connect to repository

### ISSUE THREAD

**Issue by minhongrails** [CocoaPods/CocoaPods#3142]
Currently when running "pod install" using a Podfile with proj4 (4.8.0), I get the following error:

	Installing proj4 (4.8.0)

	[!] Error installing proj4
	[!] /usr/bin/svn export --non-interactive --trust-server-cert --force "http://svn.osgeo.org/metacrs/proj/tags/4.8.0" /wrk/ios/ReplayMobile/Pods/proj4

	svn: E175002: Unable to connect to a repository at URL 'http://svn.osgeo.org/metacrs/proj/tags/4.8.0'

	svn: E175002: OPTIONS of 'http://svn.osgeo.org/metacrs/proj/tags/4.8.0': Could not read status line: connection was closed by server (http://svn.osgeo.org)

However, it appears that directory is there? I also don't see any alternative Cocoapod versions to install

**Comment by segiddins** [comment]
That looks like an `svn` error. Does running that command on it's own succeed for you?

**Comment by minhongrails** [comment]
Thanks for your fast response! Well, I tried a different project and it checked out:

	svn checkout svn://svn.code.sf.net/p/codeblocks/code/trunk 

Here are some of the commands I've tried so far:

	# same error as above
	/usr/bin/svn export --non-interactive --trust-server-cert --force "http://svn.osgeo.org/metacrs/proj/tags/4.8.0" /wrk/ios/ReplayMobile/Pods/proj4
	svn: E175002: Unable to connect to a repository at URL 'http://svn.osgeo.org/metacrs/proj/tags/4.8.0'
svn: E175002: OPTIONS of 'http://svn.osgeo.org/metacrs/proj/tags/4.8.0': Could not read status line: connection was closed by server (http://svn.osgeo.org)

	# also tried SVN protocol instead of http but no luck
	svn checkout http://svn.osgeo.org/metacrs/proj/trunk/
	svn checkout svn://svn.osgeo.org/metacrs/proj/trunk/
	svn: E000061: Unable to connect to a repository at URL 'svn://svn.osgeo.org/metacrs/proj/trunk'
svn: E000061: Can't connect to host 'svn.osgeo.org': Connection refused

I'll keep debugging but if it helps any my svn is version 1.7.17 (r1591372)

**Comment by neonichu** [comment]
FWIW, this works fine for me, using the same version of SVN (system one that ships with 10.10). Could there be any funny firewalls in place which filter specific HTTP methods?

**Comment by kylef** [comment]
Works for me with system svn too.

```
$ svn --version
svn, version 1.7.17 (r1591372)
   compiled Sep 18 2014, 13:06:44


$ /usr/bin/svn export --non-interactive --trust-server-cert --force "http://svn.osgeo.org/metacrs/proj/tags/4.8.0" proj4
...
A    proj4/proj/install-sh
Exported revision 2606.
```

The error message you had initially mentioned OPTIONS request to a certain URL, can you run this command and compare the output:

```
$ curl -X OPTIONS http://svn.osgeo.org/metacrs/proj/tags/4.8.0 -i
HTTP/1.1 200 OK
Date: Tue, 17 Feb 2015 15:50:04 GMT
Server: Apache/2.2.9 (Debian) DAV/2 SVN/1.6.9 mod_python/3.3.1 Python/2.5.2 mod_ssl/2.2.9 OpenSSL/0.9.8g
DAV: 1,2
DAV: version-control,checkout,working-resource
DAV: merge,baseline,activity,version-controlled-collection
DAV: http://subversion.tigris.org/xmlns/dav/svn/depth
DAV: http://subversion.tigris.org/xmlns/dav/svn/log-revprops
DAV: http://subversion.tigris.org/xmlns/dav/svn/partial-replay
DAV: http://subversion.tigris.org/xmlns/dav/svn/mergeinfo
DAV: <http://apache.org/dav/propset/fs/1>
MS-Author-Via: DAV
Allow: OPTIONS,GET,HEAD,POST,DELETE,TRACE,PROPFIND,PROPPATCH,COPY,MOVE,LOCK,UNLOCK,CHECKOUT
Content-Length: 0
Content-Type: text/plain
```

**Comment by minhongrails** [comment]
Thanks everyone for your responses, you'll all so awesome!

Here's the output I get from curl:

	curl -X OPTIONS http://svn.osgeo.org/metacrs/proj/tags/4.8.0 -i
	curl: (52) Empty reply from server

Based on my googling of this error, it seems that for some reason svn.osgeo.org might be terminating my OPTIONS requests on the server side or svn.osgeo.org is not resolving for some reason? Interestingly I can do GET requests just fine (I can browse http://svn.osgeo.org/metacrs/proj/tags/4.8.0 in my browser):

	curl -X GET http://svn.osgeo.org/
	<h1>OSGeo SVN Repositories</h1>
	<a href=/fdo>fdo</a><br>
	...

	# also to check DNS issues I pinged the root url, looks like it's not a DNS issue?
	ping svn.osgeo.org
	PING svn.osgeo.org (140.211.15.71): 56 data bytes
	64 bytes from 140.211.15.71: icmp_seq=0 ttl=244 time=86.255 ms
	64 bytes from 140.211.15.71: icmp_seq=1 ttl=244 time=84.155 ms

Some other things I tried were:

Updating SVN to 1.8.10 with no luck

	svn --version
	svn, version 1.8.10 (r1615264)
	   compiled Dec 10 2014, 08:22:30 on x86_64-apple-darwin14.0.0

Switching my DNS to OpenDNS (long shot)

Directly curling the IP (140.211.15.71) 

	curl -X OPTIONS 140.211.15.71/metacrs/proj/tags/4.8.0 -i
	curl: (52) Empty reply from server

Since this seems to be an issue either with my computer or network, next I'm going to try a different network (this whole time I've been on my home network with Verizon FIOS) and different computer.

**Comment by minhongrails** [comment]
It works now! :). I tested it on other machines and everything worked.

Neonichu you got it right. The issue was because of my machine and Cisco AnyConnect VPN blocking preflight OPTIONS requests. Apparently it doesn't matter if it's running or not. Once I uninstalled it all my SVN OPTIONS requests started working. I believe If I ever need to use again I should just reinstall it without the "Web Security" module. I found this solution from these pages: http://stackoverflow.com/questions/5929971/c-curl-empty-reply-from-server-every-time, http://www.bennadel.com/blog/2559-cisco-anyconnect-vpn-client-may-block-cors-ajax-options-requests.htm

I'll go ahead and close this issue since it's a user error. Thanks again everyone for your help
